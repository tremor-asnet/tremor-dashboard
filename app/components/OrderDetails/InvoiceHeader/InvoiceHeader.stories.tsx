import type { Meta, StoryObj } from "@storybook/react";

// Components
import { InvoiceHeader } from ".";
import { Card } from "@tremor/react";

// Mocks
import { mockInvoiceHeader } from "@/mocks/orderDetails";

const meta = {
  title: "Components/InvoiceHeader",
  component: InvoiceHeader,
  tags: ["autodocs"],
} as Meta<typeof InvoiceHeader>;

export default meta;

type Story = StoryObj<typeof meta>;

export const PopoverDefault: Story = {
  render: () => (
    <Card className="w-2/3 dark:bg-dark_blue">
      <InvoiceHeader {...mockInvoiceHeader} />
    </Card>
  ),
};
