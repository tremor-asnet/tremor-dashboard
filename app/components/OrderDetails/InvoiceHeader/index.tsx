import { Flex, Bold, Text, Button } from "@tremor/react";

// Types
import { InvoiceHeaderData } from "@/types";

//Styles
import "@/styles/order.css";

export const InvoiceHeader = ({
  id,
  createdAt,
  orderCode,
}: InvoiceHeaderData) => {
  return (
    <Flex className="invoice">
      <div>
        <Bold className="text-primary font-semibold capitalize dark:text-white tracking-[0.12px]">
          Order Details
        </Bold>
        <div className="mt-2 tracking-[0.4px]">
          <Text className="dark:text-dark-romance">
            Order no. <Bold>{id}</Bold> from{" "}
            <Bold>
              {createdAt.replace(/T.*/, "").split("-").reverse().join(".")}
            </Bold>
          </Text>
          <Text className="dark:text-dark-romance">
            Code: <Bold>{orderCode}</Bold>
          </Text>
        </div>
      </div>
      <Button className="py-3 px-6 bg-gradient-primary dark:bg-gradient-pickled border-none dark:text-white">
        <Text className="uppercase font-bold text-xs text-white dark:text-white">
          invoice
        </Text>
      </Button>
    </Flex>
  );
};

export default InvoiceHeader;
