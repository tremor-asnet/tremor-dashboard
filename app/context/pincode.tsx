"use client";

import {
  ReactNode,
  createContext,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from "react";

interface IPinCodeContext {
  isConfirmPinCode: boolean;
  pinCode?: number;
  isShowPinCodeModal: boolean;
  setPinCode: (code: number) => void;
  showPinCodeModal: () => void;
  hidePinCodeModal: () => void;
  confirmPinCode: (code: number) => boolean;
}

const initialPinCodeContext: IPinCodeContext = {
  isConfirmPinCode: false,
  confirmPinCode: () => false,
  hidePinCodeModal: () => {},
  setPinCode: () => false,
  showPinCodeModal: () => {},
  isShowPinCodeModal: false,
};

const PinCodeContext = createContext(initialPinCodeContext);

const PinCodeProvider = ({ children }: { children: ReactNode }) => {
  const [isConfirmPinCode, setIsConfirmPinCode] = useState(false);
  const [isShowPinCodeModal, setIsShowPinCodeModal] = useState(false);
  const [pinCode, setPinCode] = useState<number>();

  useEffect(() => {
    setIsShowPinCodeModal(!pinCode);
  }, [pinCode]);

  const handleShowPinCodeModal = useCallback(
    () => setIsShowPinCodeModal(true),
    [],
  );

  const handleHidePinCodeModal = useCallback(
    () => setIsShowPinCodeModal(false),
    [],
  );

  const handleConfirmPinCode = useCallback(
    (code: number) => {
      const isMatch = code === pinCode;
      setIsConfirmPinCode(isMatch);
      isMatch ? handleHidePinCodeModal() : handleShowPinCodeModal();
      return isMatch;
    },
    [handleHidePinCodeModal, handleShowPinCodeModal, pinCode],
  );

  const pinCodeContextValue: IPinCodeContext = {
    isConfirmPinCode,
    confirmPinCode: handleConfirmPinCode,
    pinCode,
    isShowPinCodeModal,
    setPinCode,
    hidePinCodeModal: handleHidePinCodeModal,
    showPinCodeModal: handleShowPinCodeModal,
  };

  return (
    <PinCodeContext.Provider value={pinCodeContextValue}>
      {children}
    </PinCodeContext.Provider>
  );
};

const usePinCode = () => {
  const context = useContext(PinCodeContext);

  if (!context) {
    throw new Error("usePinCode hooks should using inside PinCodeProvider!");
  }

  return context;
};

export { PinCodeProvider, usePinCode };
