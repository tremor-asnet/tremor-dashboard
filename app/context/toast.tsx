"use client";

import React, { createContext, useState } from "react";

// Components
import { FaCheckCircle } from "react-icons/fa";
import { TbExclamationMark } from "react-icons/tb";
import { RxCross2 } from "react-icons/rx";
import Toast from "@/ui/components/Toast/Toast";

export enum TOAST_TYPE {
  SUCCESS = "success",
  WARNING = "warning",
  ERROR = "error",
}
interface ToastProps {
  type?: TOAST_TYPE;
  message?: string;
  icon?: JSX.Element;
}

type ToastState = ToastProps & { isOpen: boolean };

interface ToastContextValue {
  openToast: (toastProps: ToastProps) => void;
  closeToast: () => void;
}

export const buildToastRenderer = ({ type, icon, message }: ToastProps) => {
  switch (type) {
    case TOAST_TYPE.WARNING:
      return {
        icon: icon || TbExclamationMark,
        message: message ?? "Warning!",
        color: "yellow",
      };

    case TOAST_TYPE.ERROR:
      return {
        icon: icon || RxCross2,
        message: message ?? "Error!",
        color: "red",
      };

    case TOAST_TYPE.SUCCESS:
    default:
      return {
        icon: icon || FaCheckCircle,
        message: message ?? "Success!",
        color: "green",
      };
  }
};

const ToastContext = createContext<ToastContextValue>({
  openToast: () => {},
  closeToast: () => {},
});

const ToastProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const [toast, setToast] = useState<ToastState>({
    isOpen: false,
    type: TOAST_TYPE.SUCCESS,
  });

  const closeToast = () => {
    setToast({
      ...toast,
      isOpen: false,
    });
  };

  const openToast = ({
    type = TOAST_TYPE.SUCCESS,
    message,
    icon,
  }: ToastProps) => {
    setToast({ isOpen: true, type, message, icon });

    setTimeout(() => {
      closeToast();
    }, 3000);
  };

  const toastRenderer = buildToastRenderer(toast);
  return (
    <ToastContext.Provider
      value={{
        openToast,
        closeToast,
      }}>
      {children}
      {toast.isOpen && (
        <Toast
          Icon={toastRenderer.icon}
          message={toastRenderer.message}
          color={toastRenderer.color}
          onClose={closeToast}
        />
      )}
    </ToastContext.Provider>
  );
};

export { ToastProvider, ToastContext };
