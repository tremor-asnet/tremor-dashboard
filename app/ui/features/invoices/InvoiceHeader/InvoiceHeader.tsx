"use client";

// Libs
import Link from "next/link";
import { useContext } from "react";

//Components
import { Flex, Title, Text } from "@tremor/react";
import { InvoiceLogo } from "..";

// Contexts
import { ThemeContext } from "@/context/theme";

// Themes
import { color } from "@/themes";

interface InvoiceHeaderProps {
  addressBank: string;
  cityBank: string;
  stateBank: string;
  phoneBank: string;
  fullName: string;
  addressCustomer: string;
  cityCustomer: string;
  stateCode: string;
  stateCustomer: string;
}

const InvoiceHeader = ({
  addressBank,
  cityBank,
  stateBank,
  phoneBank,
  fullName,
  addressCustomer,
  cityCustomer,
  stateCode,
  stateCustomer,
}: InvoiceHeaderProps) => {
  const { isDarkTheme } = useContext(ThemeContext);

  const renderAddressBankInfo = `${addressBank} ${cityBank}, ${stateBank}`;

  return (
    <Flex className="flex-col md:flex-row mb-10 md:mb-20">
      <Flex flexDirection="col" alignItems="start" className="md:max-w-[35%]">
        <InvoiceLogo
          color={isDarkTheme ? color.dark.primary : color.black}
          additionalClasses="browser-logo"
        />
        {/* Show this image when in print preview mode */}
        <InvoiceLogo color={color.black} additionalClasses="print-logo" />
        <Title className="text-primary dark:text-white font-semibold leading-6 tracking-wide mt-7 dark:print:text-primary">
          {renderAddressBankInfo}
        </Title>
        <Link
          href={`tel:${phoneBank}`}
          className="text-secondary print:text-black text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide mt-2 mb-6 md:mb-4 dark:print:text-black">
          tel: {phoneBank}
        </Link>
      </Flex>
      <Flex
        flexDirection="col"
        alignItems="start"
        className="md:items-end md:mt-0">
        <Title className="text-primary dark:text-white font-semibold leading-6 tracking-wide dark:print:text-primary">
          Billed to: {fullName}
        </Title>
        <Text className="text-secondary print:text-black text-tremor-title dark:text-dark-romance font-primary font-light leading-6 tracking-wide md:text-right dark:print:text-black">
          {addressCustomer}
          <br /> {cityCustomer} {stateCode}
          <br /> {stateCustomer}
        </Text>
      </Flex>
    </Flex>
  );
};

export default InvoiceHeader;
