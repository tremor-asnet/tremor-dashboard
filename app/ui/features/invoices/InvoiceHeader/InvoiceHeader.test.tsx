import { fireEvent, render } from "@testing-library/react";

// Contexts
import { ThemeContext, ThemeProvider } from "@/context/theme";

// Components
import InvoiceHeader from "./InvoiceHeader";

const InvoiceHeaderProps = {
  addressBank: "123 Bank Street",
  cityBank: "Bucharest",
  stateBank: "Romania",
  phoneBank: "+4 (074) 1090873",
  fullName: "John Doe",
  addressCustomer: "4006 Locust View Drive",
  cityCustomer: "San Francisco",
  stateCode: "CA",
  stateCustomer: "California",
};

describe("InvoiceHeader component", () => {
  it("should render snapshot correctly", () => {
    const { getByTestId, container } = render(
      <ThemeProvider>
        <ThemeContext.Consumer>
          {({ isDarkTheme, toggleTheme }) => (
            <>
              <InvoiceHeader {...InvoiceHeaderProps} />
              <button
                data-testid="toggle-theme"
                onClick={() =>
                  toggleTheme()
                }>{`isDarkTheme to ${isDarkTheme}`}</button>
            </>
          )}
        </ThemeContext.Consumer>
      </ThemeProvider>,
    );
    const themeButton = getByTestId("toggle-theme");

    fireEvent.click(themeButton);
    expect(themeButton.textContent).toBe("isDarkTheme to true");

    fireEvent.click(themeButton);
    expect(themeButton.textContent).toBe("isDarkTheme to false");

    expect(container).toMatchSnapshot();
  });

  it("should renders the component with provided props", () => {
    const { getByText } = render(<InvoiceHeader {...InvoiceHeaderProps} />);

    expect(getByText("123 Bank Street Bucharest, Romania")).toBeTruthy();
    expect(getByText("tel: +4 (074) 1090873")).toBeTruthy();
    expect(getByText("Billed to: John Doe")).toBeTruthy();
  });

  it("should correctly renders the combined address information", () => {
    const { getByText } = render(<InvoiceHeader {...InvoiceHeaderProps} />);

    expect(getByText(/4006 Locust View Drive/)).toBeTruthy();
    expect(getByText(/San Francisco CA/)).toBeTruthy();
    expect(getByText(/California/)).toBeTruthy();
  });
});
