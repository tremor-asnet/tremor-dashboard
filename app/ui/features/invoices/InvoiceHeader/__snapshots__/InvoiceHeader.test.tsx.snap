// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceHeader component should render snapshot correctly 1`] = `
<div>
  <div
    class="tremor-Flex-root flex w-full justify-between items-center flex-col md:flex-row mb-10 md:mb-20"
  >
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start md:max-w-[35%]"
    >
      <svg
        class="browser-logo"
        height="50"
        preserveAspectRatio="xMidYMid meet"
        version="1.0"
        viewBox="0 0 200.000000 200.000000"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#000"
          stroke="none"
          transform="translate(0.000000,200.000000) scale(0.100000,-0.100000)"
        >
          <path
            d="M127 1852 c-15 -17 -17 -63 -17 -454 0 -409 1 -436 18 -451 16 -15   46 -17 210 -17 l191 0 3 -399 c3 -376 4 -400 22 -420 19 -21 25 -21 451 -21   426 0 432 0 451 21 18 20 19 44 22 420 l3 399 188 0 c169 0 190 2 204 18 15   17 17 63 17 454 0 409 -1 436 -18 451 -17 16 -92 17 -874 17 -816 0 -856 -1   -871 -18z m818 -449 l0 -362 -367 -7 -368 -6 0 371 0 371 368 -2 367 -3 0   -362z m845 -4 l0 -371 -367 7 -368 6 -3 352 c-1 194 0 358 3 364 3 10 86 13   370 13 l365 0 0 -371z m-420 -839 l0 -370 -364 0 c-323 0 -365 2 -370 16 -8   21 -8 709 1 717 3 4 170 7 370 7 l363 0 0 -370z"
          />
          <path
            d="M453 1432 c-57 -54 -103 -104 -103 -111 0 -7 5 -21 10 -32 20 -36 51   -22 132 58 l78 78 78 -78 c76 -76 105 -90 130 -65 7 7 12 21 12 33 0 27 -188   215 -215 215 -12 0 -64 -42 -122 -98z"
          />
          <path
            d="M1277 1432 c-96 -96 -111 -124 -85 -150 25 -25 54 -11 130 65 l78 78   78 -78 c45 -45 86 -77 98 -77 26 0 44 18 44 44 0 27 -196 216 -224 216 -14 0   -57 -35 -119 -98z"
          />
          <path
            d="M802 668 c-27 -27 -11 -55 86 -151 71 -71 104 -97 122 -97 19 0 51   26 123 98 96 96 111 124 85 150 -24 24 -52 10 -129 -65 l-79 -77 -46 45 c-25   24 -61 59 -79 77 -34 33 -63 40 -83 20z"
          />
        </g>
      </svg>
      <svg
        class="print-logo"
        height="50"
        preserveAspectRatio="xMidYMid meet"
        version="1.0"
        viewBox="0 0 200.000000 200.000000"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="#000"
          stroke="none"
          transform="translate(0.000000,200.000000) scale(0.100000,-0.100000)"
        >
          <path
            d="M127 1852 c-15 -17 -17 -63 -17 -454 0 -409 1 -436 18 -451 16 -15   46 -17 210 -17 l191 0 3 -399 c3 -376 4 -400 22 -420 19 -21 25 -21 451 -21   426 0 432 0 451 21 18 20 19 44 22 420 l3 399 188 0 c169 0 190 2 204 18 15   17 17 63 17 454 0 409 -1 436 -18 451 -17 16 -92 17 -874 17 -816 0 -856 -1   -871 -18z m818 -449 l0 -362 -367 -7 -368 -6 0 371 0 371 368 -2 367 -3 0   -362z m845 -4 l0 -371 -367 7 -368 6 -3 352 c-1 194 0 358 3 364 3 10 86 13   370 13 l365 0 0 -371z m-420 -839 l0 -370 -364 0 c-323 0 -365 2 -370 16 -8   21 -8 709 1 717 3 4 170 7 370 7 l363 0 0 -370z"
          />
          <path
            d="M453 1432 c-57 -54 -103 -104 -103 -111 0 -7 5 -21 10 -32 20 -36 51   -22 132 58 l78 78 78 -78 c76 -76 105 -90 130 -65 7 7 12 21 12 33 0 27 -188   215 -215 215 -12 0 -64 -42 -122 -98z"
          />
          <path
            d="M1277 1432 c-96 -96 -111 -124 -85 -150 25 -25 54 -11 130 65 l78 78   78 -78 c45 -45 86 -77 98 -77 26 0 44 18 44 44 0 27 -196 216 -224 216 -14 0   -57 -35 -119 -98z"
          />
          <path
            d="M802 668 c-27 -27 -11 -55 86 -151 71 -71 104 -97 122 -97 19 0 51   26 123 98 96 96 111 124 85 150 -24 24 -52 10 -129 -65 l-79 -77 -46 45 c-25   24 -61 59 -79 77 -34 33 -63 40 -83 20z"
          />
        </g>
      </svg>
      <p
        class="text-tremor-title text-primary dark:text-white font-semibold leading-6 tracking-wide mt-7 dark:print:text-primary"
      >
        123 Bank Street Bucharest, Romania
      </p>
      <a
        class="text-secondary print:text-black text-base dark:text-dark-romance font-primary font-light leading-6 tracking-wide mt-2 mb-6 md:mb-4 dark:print:text-black"
        href="tel:+4 (074) 1090873"
      >
        tel: 
        +4 (074) 1090873
      </a>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start md:items-end md:mt-0"
    >
      <p
        class="text-tremor-title text-primary dark:text-white font-semibold leading-6 tracking-wide dark:print:text-primary"
      >
        Billed to: 
        John Doe
      </p>
      <p
        class="text-secondary print:text-black text-tremor-title dark:text-dark-romance font-primary font-light leading-6 tracking-wide md:text-right dark:print:text-black"
      >
        4006 Locust View Drive
        <br />
         
        San Francisco
         
        CA
        <br />
         
        California
      </p>
    </div>
  </div>
  <button
    data-testid="toggle-theme"
  >
    isDarkTheme to false
  </button>
</div>
`;
